logstreams/src/test/java/io/zeebe/logstreams/processor/AsyncSnapshotingTest.java

```diff
@@ -135,11 +137,75 @@ public void shouldValidSnapshotWhenCommitPositionGreaterEquals() throws Exceptio
     inOrder.verifyNoMoreInteractions();
   }
 
+  @Test
+  public void shouldNotTakeSameSnapshotTwice() throws Exception {
+    // given
+    when(positionSupplier.get()).thenReturn(CompletableActorFuture.completed(25L));
+    logStreamRule.getClock().addTime(Duration.ofMinutes(1));
+    logStreamRule.setCommitPosition(100L);
+    verify(snapshotController, timeout(TIMEOUT).times(1)).moveValidSnapshot(25);
+    verify(positionSupplier, timeout(TIMEOUT).times(1)).get();
+
+    // when
+    logStreamRule.getClock().addTime(Duration.ofMinutes(1));
+    verify(positionSupplier, timeout(TIMEOUT).times(2)).get();
+
+    // then
+    final InOrder inOrder = Mockito.inOrder(snapshotController);
+    inOrder.verify(snapshotController, timeout(TIMEOUT).times(1)).takeTempSnapshot();
+    inOrder.verify(snapshotController, timeout(TIMEOUT).times(1)).moveValidSnapshot(25);
+    inOrder.verify(snapshotController, timeout(TIMEOUT).times(1)).ensureMaxSnapshotCount(3);
+    inOrder.verify(snapshotController, timeout(TIMEOUT).times(1)).replicateLatestSnapshot(any());
+    inOrder.verifyNoMoreInteractions();
+  }
+
+  @Test
+  public void shouldNotFailOnFailingReplication() throws Exception {
```

Comment:
I think we can remove this test case because it is also tested in `shouldNotStopTakingSnapshotsAfterFailingReplication()`.